docs/src/pages/reference/configuration-reference.md

```diff
@@ -3,22 +3,44 @@ layout: ~/layouts/MainLayout.astro
 title: Configuration Reference
 ---
 
-To configure Astro, add an `astro.config.mjs` file in the root of your project. All settings are optional.
-
-You can view the full configuration API (including information about default configuration) on [GitHub.](https://github.com/withastro/astro/blob/latest/packages/astro/src/%40types/astro.ts)
+To configure Astro, add an `astro.config.mjs` file to the root of your project.
 
 ```js
-// Example: astro.config.mjs
-
-// @type-check enabled!
-// VSCode and other TypeScript-enabled text editors will provide auto-completion,
-// helpful tooltips, and warnings if your exported object is invalid.
-// You can disable this by removing "@ts-check" and `@type` comments below.
-
-// @ts-check
-export default /** @type {import('astro').AstroUserConfig} */ (
-  {
-    // ...
-  }
-);
+export default /** @type {import('astro').AstroUserConfig} */ ({
+  projectRoot: './',
+  public: './public/',
+  dist: './dist/',
+  src: './src/',
+  pages: './pages/',
+  renderers: [
+    '@astrojs/renderer-svelte',
+    '@astrojs/renderer-vue',
+    '@astrojs/renderer-react',
+    '@astrojs/renderer-preact',
+  ],
+});
 ```
+
+All settings are optional.
+
+The `projectRoot` option sets the working directory used by Astro. All other paths will be resolved from this. By default, this is the current working directory.
+
+The `public` option sets the directory used to resolve public assets. By default, this is the `public` directory.
+
+The `dist` option sets the directory used to write the production build of the project.
+
+The `src` option sets the directory used to resolve source files, like `pages`. By default, this is the `src` directory.
+
+The `pages` option sets the directory used to resolve pages, relative to the `src` option. By default, this is the `pages` directory.
+
+The `renderers` option assigns the framework renderers to be used by Astro. By default, Astro provides renderers for Svelte, Vue, React, and Preact. To assign no renderers at all, provide an empty array (`[]`).
```

Comment:
```suggestion
The `renderers` opt-in option assigns the relevant framework renderers to be used by Astro. By default, Astro provides renderers for Svelte, Vue, React, and Preact. To assign no renderers at all, provide an empty array (`[]`).
```

Reply:
Tiny comment to say that I think the phrase "By default, Astro provides..." is doing a lot of work here. 

I could see this being interpreted as "included in your project out of the box" (especially in conjunction with "default" which makes you think of the state you'll find when you first initialize a new project) vs what I think is more accurately "available to be added to your project." 

Would Fuzzy be happier with leaving the first sentence alone, but then something like:

"Astro *optionally* allows you to specify renderers for ... To assign no renderers at all, provide an empty array (`[]`) or comment out the field entirely." <-- assuming it's not commented out to begin with.

Carry on!

Reply:
I’ve rewritten all sections to avoid "By default" doing so much work.